site/documentation/content/api/management/device-registry-v1.yaml

```diff
@@ -620,7 +646,76 @@ components:
          description: Allows arbitrary properties as extension to the ones
                       specified by the Hono API.
 
-# Tenant schema
+
+      PagingSortingFilteringOptions:
+         type: object
+         additionalProperties: false
+         properties:
+            "page":
+               $ref: '#/components/schemas/PagingOptions'
+            "filters":
+               $ref: '#/components/schemas/FilterOptionsList'
+            "sort":
+               $ref: '#/components/schemas/SortingOptions'
+
+      PagingOptions:
+         type: object
+         additionalProperties: false
+         description: options to page the results of bulk queries.
+         properties:
+            "limit":
+               type: integer
+               default: 30
+               minimum: 0
+               maximum: 200
+               description: The maximum number of objects to include in a response.
+            "offset":
+               type: integer
+               default: 0
+               minimum: 0
+               description: |
+                  The offset to use as a starting point to return results. This allows to go through the query results in
+                  a paged manner when the query returns more results than the limit.
+
+      FilterOptionsList:
+         type: array
+         description: A list of filter expressions. Objects must match all filters in order to be included in a result.
+         items:
+            $ref: '#/components/schemas/FilterOption'
+
+      FilterOption:
+            type: object
+            additionalProperties: false
+            description: Filters to be applied to bulk queries.
+            properties:
+               "field":
+                  type: string
+                  description: The JSON Path representation of the field to use for filtering.
+               "value":
+                  type: string
+                  description: The value to filter the field with.
+            examples:
+               Filters options:
+                  $ref: '#/components/examples/FilterOptionsExample'
+
+      SortingOptions:
```

Comment:
this object defines a single sort option only, doesn't it?
If you want to allow for multiple sort options to be given, then the `PagingSortingFilteringOptions.sort` property needs to be an array, right?

Reply:
IMHO having multiples sort options at the same time is a bit weird (from a UI/UX perspective) but i guess it doesn't hurt to support it in the back end.

Reply:
I am in no way arguing that we *must* have multiple sorting options. We could as well have just one. I only want to make sure that we know what we want/get :-) So, supporting only one sort option for the time being is fine with me. In any case, the object should be named `SortingOption` as it only defines a single property and sort direction, right? If we want to support multiple options later, then we would allow specifying an array of `SortingOption` instances in `PagingSortingFilteringOptions`. In that case (multiple options) my understanding would be that the results are ordered following the order of the `SortingOption` instances in the array ...

Reply:
I am fine with both as well. Sorry if my comment came across as arguing, I wasn't either. 
I am thinking that having an array to support multiples options is the most flexible. The underlying implementation can choose to only support one and apply the first element of the array.

Reply:
I pushed a change using multiples filters and improved the documentation.